contracts/test/StakeTable.t.sol

```diff
@@ -838,3 +842,359 @@ contract StakeTableUpgradeTest is Test {
+        timelockController.execute(address(proxyAddress), 0, data, bytes32(0), bytes32(0));
+        vm.stopPrank();
+        vm.assertFalse(timelockController.isOperationDone(txId));
+    }
+
+    function test_RevertWhen_ExecutionWithoutCorrectPermission() public {
+        vm.startPrank(makeAddr("notProposer"));
+
+        // Encode upgrade call
+        bytes memory data =
+            abi.encodeWithSignature("upgradeToAndCall(address,bytes)", address(new S()), "");
+
+        bytes32 txId =
+            timelockController.hashOperation(address(stakeTable), 0, data, bytes32(0), bytes32(0));
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IAccessControl.AccessControlUnauthorizedAccount.selector,
+                address(makeAddr("notProposer")),
+                timelockController.PROPOSER_ROLE()
+            )
+        );
+        timelockController.schedule(address(stakeTable), 0, data, bytes32(0), bytes32(0), DELAY);
+        vm.stopPrank();
+
+        vm.startPrank(proposers[0]);
+        timelockController.schedule(address(stakeTable), 0, data, bytes32(0), bytes32(0), DELAY);
+        vm.stopPrank();
+
+        vm.warp(block.timestamp + DELAY + 1);
+
+        vm.startPrank(makeAddr("notExecutor"));
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IAccessControl.AccessControlUnauthorizedAccount.selector,
+                address(makeAddr("notExecutor")),
+                timelockController.EXECUTOR_ROLE()
+            )
+        );
+        timelockController.execute(address(proxyAddress), 0, data, bytes32(0), bytes32(0));
+        vm.stopPrank();
+        vm.assertFalse(timelockController.isOperationDone(txId));
+    }
+
+    function test_RevertWhen_ExecuteWithWrongSalt() public {
+        // Encode upgrade call
+        bytes memory data =
+            abi.encodeWithSignature("upgradeToAndCall(address,bytes)", address(new S()), "");
+
+        bytes32 correctSalt = keccak256("salt-A");
+        bytes32 wrongSalt = keccak256("salt-B");
+
+        bytes32 wrongTxId =
+            timelockController.hashOperation(address(stakeTable), 0, data, wrongSalt, bytes32(0));
+        vm.startPrank(proposers[0]);
+        timelockController.schedule(address(stakeTable), 0, data, correctSalt, bytes32(0), DELAY);
+        vm.stopPrank();
+
+        vm.warp(block.timestamp + DELAY + 1);
+
+        vm.startPrank(executors[0]);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                TimelockController.TimelockUnexpectedOperationState.selector,
+                wrongTxId,
+                bytes32(1 << uint8(TimelockController.OperationState.Ready))
+            )
+        );
+        timelockController.execute(address(stakeTable), 0, data, wrongSalt, bytes32(0));
+        vm.stopPrank();
+    }
+
+    function test_UnauthorizedCannotUpgrade() public {
+        vm.startPrank(makeAddr("notAdmin"));
+
+        try stakeTable.upgradeToAndCall(address(new S()), "") {
```

Comment:
Similar to a previous PR, do we need `try / catch` only because we have a deployment inside the function argument statement?